tests/rptest/tests/rbac_test.py

```diff
@@ -120,37 +138,115 @@ def test_regular_user_access(self):
+
+        self.superuser_admin.create_role(role=role_one)
+        self.superuser_admin.create_role(role=role_two)
+        self.superuser_admin.create_role(role=role_three)
+
+        def matching_role_set(filter: Optional[str] = None,
+                              principal: Optional[str] = None,
+                              principal_type: Optional[str] = None):
+            res = self.superuser_admin.list_roles(
+                filter=filter,
+                principal=principal,
+                principal_type=principal_type)
+            roles = RolesList.from_response(res).roles
+            return set(role.name for role in roles)
+
+        def test_filter(expected_roles: set[str],
+                        filter: Optional[str] = None,
+                        principal: Optional[str] = None,
+                        principal_type: Optional[str] = None):
+            wait_until(lambda: set(expected_roles) == \
+                                matching_role_set(filter=filter, principal=principal, principal_type=principal_type),
+                       timeout_sec=10,
+                       err_msg=f"Mismatch on filter result")
+
+        self.logger.debug(
+            "Test that list_roles works with various combinations of filter")
+        test_filter(filter="aaaa", expected_roles=[role_one, role_two])
+        test_filter(filter="bb", expected_roles=[role_three])
+        test_filter(filter="aaaa-2", expected_roles=[role_two])
+        test_filter(filter="ccc", expected_roles=[])
+
+        self.logger.debug(
+            "Test that list_roles works with an unknown principal")
+        test_filter(principal=alice.name, expected_roles=[])
+
+        self.logger.debug(
+            "Test that list_roles works with an existing principals")
+        self.superuser_admin.update_role_members(role=role_one, add=[alice])
+        self.superuser_admin.update_role_members(role=role_two,
+                                                 add=[alice, bob])
+        self.superuser_admin.update_role_members(role=role_three, add=[bob])
+
+        test_filter(principal=alice.name, expected_roles=[role_one, role_two])
+        test_filter(principal=bob.name, expected_roles=[role_two, role_three])
+        test_filter(filter="aaaa",
+                    principal=bob.name,
+                    expected_roles=[role_two])
+
+        self.logger.debug(
+            "Test that list_roles rejects a non-User principal type")
+        with expect_role_error(RoleErrorCode.MALFORMED_DEF):
+            self.superuser_admin.list_roles(filter="aaa",
+                                            principal=alice.name,
+                                            principal_type="Role")
+        self.logger.debug("Test that list_roles accepts a User principal type")
+        test_filter(filter="aaaa",
+                    principal=bob.name,
+                    principal_type="User",
+                    expected_roles=[role_two])
+
```

Comment:
suggestion (if-minor): Could add a sanity check that default principal_type works as expected, e.g

```python
        test_filter(filter="aaaa",
                    principal=bob.name,
                    expected_roles=[role_two])
```

Should be a well-formed request.

Reply:
I have exactly this on line 234 :)